base_layer/contacts/src/contacts_service/storage/types/messages.rs

```diff
@@ -83,14 +85,42 @@ impl MessagesSqlInsert {
 
 impl MessagesSql {
     /// Find a particular message by their address, if it exists
+    pub fn find_by_to_address(
+        address: &[u8],
+        limit: i64,
+        page: i64,
+        conn: &mut SqliteConnection,
+    ) -> Result<Vec<MessagesSql>, ContactsServiceStorageError> {
+        Ok(messages::table
+            .filter(messages::to_address.eq(address))
+            .order(messages::stored_at.desc())
+            .offset(limit * page)
+            .limit(limit)
+            .load::<MessagesSql>(conn)?)
+    }
+
+    pub fn find_by_from_address(
+        address: &[u8],
+        limit: i64,
+        page: i64,
+        conn: &mut SqliteConnection,
+    ) -> Result<Vec<MessagesSql>, ContactsServiceStorageError> {
+        Ok(messages::table
+            .filter(messages::from_address.eq(address))
+            .order(messages::stored_at.desc())
+            .offset(limit * page)
+            .limit(limit)
+            .load::<MessagesSql>(conn)?)
+    }
```

Comment:
This interface makes it a bit difficult for ffi clients.

Previously you could get an entire message history of sent and received messages by querying the address of the contact (other party). Messages were distinguished by the `direction` to know if they were outbound or inbound. 

This new interface now requires two function calls, and a merging of the results, sorted by received dates to generate a timeline of messages.

I think the interface we're seeking isn't to find by `from` or `to` address independently, but to find by address where the address given is in *either* the `from` or `to` column. 
To put it plainly. It should still be a single function that checks if the address is in either field, but we don't actually care which one it belongs to.

We may also be able to remove the `direction` field as it becomes redundant if we're maintaining the to/from address and in all cases one of those fields will always be the current node. Direction becomes something derivable by which field our address is in.

Reply:
I dont think these are even mapped to the FFI atm, 
Currently they still use `find_by_address`
This is now changed to do a `.filter(messages::from_address.eq(address).or(messages::to_address.eq(address)))`
So you can still get a complete history by just doing a simple lookup to an address.

Reply:
ahhhh i thought the old method had been removed